pkg/apis/v1beta1/taints.go

```diff
@@ -32,6 +33,21 @@ var (
 		Effect: v1.TaintEffectNoSchedule,
 		Value:  DisruptingNoScheduleTaintValue,
 	}
+
+	DisruptionCandidatePreferNoScheduleTaint = v1.Taint{
+		Key:    DisruptionCandidateTaintKey,
+		Effect: v1.TaintEffectPreferNoSchedule,
+	}
+
+	// DisruptionCandidateNoScheduleTaint is applied in-memory to drifted and expired nodes.
+	// This, along with the PreferNoSchedule taint applied to all disruption candidates, ensures
+	// that we don't reschedule nodes on disruption candidates to other disruption candidates.
+	// Note: this taint is **only** applied in-memory. It does not get applied to nodes on the
+	// API server.
+	DisruptionCandidateNoScheduleTaint = v1.Taint{
```

Comment:
I'm still not convinced that we want to store this in the APIs package. This isn't used externally by anything. Why not in scheduling?

Reply:
I'm good with removing it and constructing the taint directly in statenode.